src/libguestfish.sh

```diff
@@ -0,0 +1,48 @@
+# Helper library for using libguestfs on CoreOS-style images.
+# A major assumption here is that the disk image uses OSTree
+# and also has `boot` and `root` filesystem labels.
+
+# http://libguestfs.org/guestfish.1.html#using-remote-control-robustly-from-shell-scripts
+GUESTFISH_PID=
+coreos_gf_launch() {
+    local guestfish
+    guestfish[0]="guestfish"
+    guestfish[1]="--listen"
+    guestfish[3]="-a"
+    guestfish[4]="${tmp_dest}"
+
+    eval $("${guestfish[@]}")
+    if [ -z "$GUESTFISH_PID" ]; then
+        fatal "guestfish didn't start up, see error messages above"
+    fi
+}
+
+_coreos_gf_cleanup () {
+    guestfish --remote -- exit >/dev/null 2>&1 ||:
+}
+trap _coreos_gf_cleanup EXIT ERR
```

Comment:
Not new, but `EXIT` should be all we need here. IIRC, `ERR` only gets triggered when exiting due to `set -e`. `EXIT` gets triggered on exit, regardless of why.

Reply:
This is from http://libguestfs.org/guestfish.1.html - anyways I changed it.